cfme/tests/integration/test_menu_plugin.py

```diff
@@ -0,0 +1,79 @@
+import pytest
+import cfme.configure.access_control as ac
+import fauxfactory
+from cfme import Credential, login
+from utils import menu_plugin as mp
+from cfme.fixtures import pytest_selenium as sel
+from utils.providers import setup_a_provider
+
+
+@pytest.fixture(scope="module")
+def setup_infra_provider():
+    return setup_a_provider(prov_class="infra", validate=True, check_existing=True,
+        required_keys=['ownership_vm'])
+
+
+@pytest.fixture(scope="module")
+def setup_menu_plugin():
+    mp.menu_plugin_setup()
+
+
+@pytest.yield_fixture(scope="module")
+def new_role():
+    role = ac.Role(name='role_' + fauxfactory.gen_alphanumeric())
+    role.create()
+    yield role
+    login.login_admin()
+    role.delete()
+
+
+@pytest.yield_fixture(scope="module")
+def new_group(new_role):
+    group = ac.Group(description='group_' + fauxfactory.gen_alphanumeric(),
+        role=new_role.name)
+    group.create()
+    yield group
+    login.login_admin()
+    group.delete()
+
+
+def new_credential():
+    return Credential(principal='uid' + fauxfactory.gen_alphanumeric(), secret='redhat')
+
+
+@pytest.yield_fixture(scope="module")
+def non_admin_user(new_group):
+    user1 = ac.User(name='user_' + fauxfactory.gen_alphanumeric(),
+                credential=new_credential(),
+                email='abc@redhat.com',
+                group=new_group,
+                cost_center='Workload',
+                value_assign='Database')
+    user1.create()
+    yield user1
+    login.login_admin()
+    user1.delete()
+
+
+@pytest.mark.meta(blockers=["BZ#1286627"])
+def test_default_module(setup_menu_plugin):
+    # Click Redhat Homepage link
+    sel.force_navigate('rh_homepage')
+    mp.menu_plugin_assert()
```

Comment:
Just thinking about this......I think we are unfortunately going to have to do something a little different, simply because are using the same assertion function twice. However we expect those pages to be different. We should expect the assertion to be different too. This goes back to whether we should use two internal pages, that I mentioned earlier. I think we should, and perhaps those URLs should be stored in the creds yamls, the files loaded and substituted and then pushed up to the appliance.